src/tests/Loader/async/async2gcRootsScan.cs

```diff
@@ -0,0 +1,159 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using Xunit;
+
+public class Async2RootReporting
+{
+    private static TaskCompletionSource<int> cs;
+
+
+    static async Task<int> Recursive1(int n)
+    {
+        Task<int> cTask = cs.Task;
+
+        // make some garbage
+        object o1 = n;
+        //object o2 = n;
+        //object o3 = n;
+        //object o4 = n;
+        //object o5 = n;
+        //object o6 = n;
+        //object o7 = n;
+        //object o8 = n;
+        //object o9 = n;
+        //object o10 = n;
+        //object o11 = n;
+        //object o12 = n;
+
+        if (n == 0)
+            return await cTask;
+
+        var result = await Recursive1(n - 1);
+
+        Assert.Equal(n, (int)o1);
+        //Assert.Equal(n, (int)o2);
+        //Assert.Equal(n, (int)o3);
+        //Assert.Equal(n, (int)o4);
+        //Assert.Equal(n, (int)o5);
+        //Assert.Equal(n, (int)o6);
+        //Assert.Equal(n, (int)o7);
+        //Assert.Equal(n, (int)o8);
+        //Assert.Equal(n, (int)o9);
+        //Assert.Equal(n, (int)o10);
+        //Assert.Equal(n, (int)o11);
+        //Assert.Equal(n, (int)o12);
+
+        return result;
+    }
+
+    static async2 Task<int> Recursive2(int n)
+    {
+        Task<int> cTask = cs.Task;
+
+        // make some garbage
+        object o1 = n;
+
+// THIS CAUSES CRASHES IN UNWINDER FLAVOR
+        //object o2 = n;
+        //object o3 = n;
```

Comment:
I'd like to uncomment these to have a more real benchmark, but it causes crashes in unwinder-based implementation. 
I think it can be looked at separately from this change.